test/uk/gov/hmrc/transitmovementsvalidator/controllers/MessagesControllerSpec.scala

```diff
@@ -309,4 +305,104 @@ class MessagesControllerSpec
     }
   }
 
+  "On validate Object Store resource" - {
+
+    lazy val validSource = Source.single(ByteString(<test>test xml</test>.mkString, StandardCharsets.UTF_8))
+    "on a valid file being streamed from object store with X-Object-Store-Uri present and no content type, return No Content" in {
+      when(mockXmlValidationService.validate(eqTo(validCode), any[Source[ByteString, _]])(any[Materializer], any[ExecutionContext]))
+        .thenAnswer(
+          _ => EitherT.rightT[Future, ValidationError](())
+        )
+
+      when(mockObjectStoreService.getContents(eqTo("http://hmrc.gov.uk/filename"))(any[ExecutionContext], any[HeaderCarrier]))
+        .thenAnswer(
+          _ => EitherT.rightT[Future, Source[ByteString, NotUsed]](validSource)
+        )
+
+      val request =
+        FakeRequest("POST", s"/messages/$validCode/validate/", FakeHeaders(Seq("X-Object-Store-Uri" -> "http://hmrc.gov.uk/filename")), Source.empty)
+      val sut    = new MessagesController(stubControllerComponents(), mockXmlValidationService, mockJsonValidationService, mockObjectStoreService)
+      val result = sut.validate(validCode)(request)
+
+      status(result) mustBe NO_CONTENT
+    }
+
+    "on a file being streamed from object store with neither X-Object-Store-Uri nor content type present, return Unsupported Media Type with an error message" in {
+      val sut = new MessagesController(stubControllerComponents(), mockXmlValidationService, mockJsonValidationService, mockObjectStoreService)
+      val request =
+        FakeRequest("POST", s"/messages/$validCode/validate/", FakeHeaders(), Source.empty)
+      val result = sut.validate(validCode)(request)
+
+      contentAsJson(result) mustBe Json.obj("message" -> "Content Type or X-Object-Store-Uri must be specified.", "code" -> "UNSUPPORTED_MEDIA_TYPE")
```

Comment:
This should be bad request error.

Reply:
After discussion will change to Bad Request.